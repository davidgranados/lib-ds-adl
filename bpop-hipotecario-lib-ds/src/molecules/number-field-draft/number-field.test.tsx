import React from "react";
import { mount, ReactWrapper } from "enzyme";
import NumberField from "./number-field";

describe("<NumberField />", () => {
  let wrapper: ReactWrapper<typeof NumberField>;
  const props = {
    id: "test-number-field-id",
    label: "test number field label",
    maxLength: 10,
    caption: "test caption",
    placeholder: "test placeholder",
    size: "md",
    onFocus: jest.fn(),
    onBlur: jest.fn(),
    onChange: jest.fn(),
    onInput: jest.fn(),
  };
  describe("<NumberField /> with initial value", () => {
    const initialValue = 123;
    beforeEach(() => {
      jest.clearAllMocks();
      // given
      wrapper = mount<typeof NumberField>(<NumberField {...props} value={initialValue} />);
    });
    it("should match snapshot", () => {
      // then
      expect(wrapper).toMatchSnapshot();
    });
    describe("<Label />", () => {
      it("should render correctly", () => {
        // when
        const labelElement = wrapper.find("label");
        // then
        expect(labelElement.prop("htmlFor")).toEqual(props.id);
        expect(labelElement.prop("children")).toEqual(props.label);
      });
    });
    describe("<InputBase />", () => {
      it("should render correctly", () => {
        // when
        const inputElement = wrapper.find("input");
        // then
        expect(inputElement.prop("id")).toEqual(props.id);
        expect(inputElement.prop("value")).toEqual(initialValue);
        expect(inputElement.prop("type")).toEqual("tel");
        expect(inputElement.prop("placeholder")).toEqual(props.placeholder);
        expect(inputElement.hasClass(`size-${props.size}`)).toBe(true);
        expect(inputElement.hasClass(`filled`)).toBe(true);
      });
    });
  });
  describe("<NumberField /> with empty value", () => {
    beforeEach(() => {
      jest.clearAllMocks();
      // given
      wrapper = mount<typeof NumberField>(<NumberField {...props} />);
    });
    it("should match snapshot", () => {
      // then
      expect(wrapper).toMatchSnapshot();
    });
    describe("<InputBase />", () => {
      it("should render correctly", () => {
        // when
        const inputElement = wrapper.find("input");
        // then
        expect(inputElement.prop("value")).toEqual("");
      });
    });
  });
  describe("<NumberField /> with error caption", () => {
    beforeEach(() => {
      jest.clearAllMocks();
      // given
      wrapper = mount<typeof NumberField>(<NumberField {...props} status={"error"} />);
    });
    it("should match snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
    describe("<InputBase />", () => {
      it("should have error class", () => {
        // when
        const inputElement = wrapper.find("input");
        // then
        expect(inputElement.hasClass("error")).toBe(true);
      });
    });
    describe("<Typography />", () => {
      it("should render correctly", () => {
        // when
        const labelElement = wrapper.find("span");
        // then
        expect(labelElement.prop("children")).toEqual(props.caption);
      });
    });
  });
  describe("<NumberField /> change event", () => {
    beforeEach(() => {
      jest.clearAllMocks();
      // given
      wrapper = mount<typeof NumberField>(<NumberField {...props} />);
    });
    it("should handle change", () => {
      // when
      wrapper.find("input").simulate("change");
      // then
      expect(props.onChange).toHaveBeenCalled();
    });
    it("should accept numbers", () => {
      // given
      const mockChangeEvent = { target: { value: 456 } };
      // when
      wrapper.find("input").simulate("change", mockChangeEvent);
      // then
      expect(wrapper.find("input").prop("value")).toEqual(mockChangeEvent.target.value);
    });
    it("shouldn't accept letters or special characters", () => {
      // given
      const mockChangeEvent = { target: { value: "test" } };
      // when
      wrapper.find("input").simulate("change", mockChangeEvent);
      // then
      expect(wrapper.find("input").prop("value")).toEqual("");
    });
  });
});
