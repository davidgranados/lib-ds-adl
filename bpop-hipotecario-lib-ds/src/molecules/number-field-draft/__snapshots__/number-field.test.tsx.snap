// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberField /> <NumberField /> with empty value should match snapshot 1`] = `
<div
  className="wrapper"
>
  <label
    className="default"
    htmlFor="test-number-field-id"
  >
    test number field label
  </label>
  <input
    className="default size-md"
    id="test-number-field-id"
    maxLength={10}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onInput={[Function]}
    placeholder="test placeholder"
    type="tel"
    value=""
  />
</div>
`;

exports[`<NumberField /> <NumberField /> with error caption should match snapshot 1`] = `
<div
  className="wrapper"
>
  <label
    className="default"
    htmlFor="test-number-field-id"
  >
    test number field label
  </label>
  <input
    className="default size-md error"
    id="test-number-field-id"
    maxLength={10}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onInput={[Function]}
    placeholder="test placeholder"
    type="tel"
    value=""
  />
  <span
    className="default body-sm weight-regular error-600 line-height-body-sm"
  >
    test caption
  </span>
</div>
`;

exports[`<NumberField /> <NumberField /> with initial value should match snapshot 1`] = `
<div
  className="wrapper"
>
  <label
    className="default"
    htmlFor="test-number-field-id"
  >
    test number field label
  </label>
  <input
    className="default size-md filled"
    id="test-number-field-id"
    maxLength={10}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onInput={[Function]}
    placeholder="test placeholder"
    type="tel"
    value={123}
  />
</div>
`;
